query/physicalplan/filter.go

```diff
@@ -142,6 +142,52 @@ func binaryBooleanExpr(expr *logicalplan.BinaryExpr) (BooleanExpression, error)
 			Left:  left,
 			Right: right,
 		}, nil
+	case logicalplan.OpAdd, logicalplan.OpSub, logicalplan.OpMul, logicalplan.OpDiv:
+		var leftColumnRef *ArrayRef
+		var leftScalar scalar.Scalar
+		expr.Left.Accept(PreExprVisitorFunc(func(expr logicalplan.Expr) bool {
+			switch e := expr.(type) {
+			case *logicalplan.LiteralExpr:
+				leftScalar = e.Value
+				return false
+			case *logicalplan.Column:
+				leftColumnRef = &ArrayRef{
+					ColumnName: e.ColumnName,
+				}
+				return false
+			}
+			return true
+		}))
+		if leftColumnRef == nil && leftScalar == nil {
+			return nil, errors.New("left side of arithmetic expression must be a column or a scalar")
+		}
+
+		var rightColumnRef *ArrayRef
+		var rightScalar scalar.Scalar
+		expr.Right.Accept(PreExprVisitorFunc(func(expr logicalplan.Expr) bool {
+			switch e := expr.(type) {
+			case *logicalplan.LiteralExpr:
+				rightScalar = e.Value
+				return false
+			case *logicalplan.Column:
+				rightColumnRef = &ArrayRef{
+					ColumnName: e.ColumnName,
+				}
+				return false
+			}
+			return true
+		}))
+		if rightColumnRef == nil && rightScalar == nil {
+			return nil, errors.New("right side of arithmetic expression must be a column or a scalar")
+		}
+
+		return &ArithmeticExpr{
+			left:        leftColumnRef,
+			leftScalar:  leftScalar,
+			right:       rightColumnRef,
+			rightScalar: rightScalar,
+			operation:   expr.Op.String(),
```

Comment:
Why use the string representation instead of the `Op` itself since it's an enum?